qa/embedded/src/test/java/io/camunda/zeebe/process/test/qa/embedded/injection/InheritanceInjectionTest.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright © 2021 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.camunda.zeebe.process.test.qa.embedded.injection;
+
+import io.camunda.zeebe.process.test.extension.ZeebeProcessTest;
+import io.camunda.zeebe.process.test.qa.abstracts.injection.AbstractInheritanceInjectionTest;
+
+@ZeebeProcessTest
```

Comment:
:thought_balloon: I wonder if the `@ZeebeProcessTest` annotation should be placed on the `AbstractInheritanceInjectionTest`, as this is how it was reported in the issue. I tested it manually, and it works.

:thinking: On the other hand, it makes more sense to use the `@ZeebeProcessTest` annotation on the class where all the tests are.

:+1: In any case, the new code will work, and is properly tested.

Reply:
Good question. I did think about this and decided against it in order to prevent duplication in our tests.

As you noticed we have 2 extensions in this project. One is for using an embedded Zeebe engine, the other is for using testcontainers. Depending on the Java version users can't use the embedded version, as it is tied to the Java version we use in Zeebe.

Because I didn't want to duplicate all testcases we have this "strange" test structure where we have all our tests defined in an abstract test class. Each of these abstract classes is inherited by 2 other test classes, 1 for the embedded extension and 1 for the testcontainers extension. The subclasses only provide the relevant extension to run the test using either the embedded engine, or the testcontainers engine.

As a result if we want to put the extension on the superclass, we would have to move the implementation of this test to `embedded` and the `testcontainers` module so we can test it for both extensions. As a result, we have a duplicated testcase 🙂

Reply:
:+1: Then it makes sense to leave it as it is. The test is still correct, plus it avoids duplication.